crawler.py

```diff
@@ -111,10 +112,23 @@
 
 unsaved_jobs = list()
 unsaved_jobs_lock = Lock()
+
+
+def mark_job_done(logger: logging.Logger, unsaved_jobs_lock: Lock,
```

Comment:
I think it's confusing to still have this called `mark_job_done`, since it implies that calling this function marks the job "done". Can we change to `get_job_completion_callback` or something along those lines?